tests/python_package_test/test_basic.py

```diff
@@ -217,6 +217,43 @@ def test_sequence_get_data(num_seq):
     np.testing.assert_array_equal(subset_data.get_data(), X[sorted(used_indices)])
 
 
+def test_retrain_list_of_sequence():
+    X, y = load_breast_cancer(return_X_y=True)
+    seqs = _create_sequence_from_ndarray(X, 2, 100)
+
+    seq_ds = lgb.Dataset(seqs, label=y, free_raw_data=False)
+
+    assert sum([len(s) for s in seq_ds.get_data()]) == X.shape[0]
+    assert len(seq_ds.get_feature_name()) == X.shape[1]
+    assert seq_ds.get_data() == seqs
+
+    params = {
+        "objective": "binary",
+        "num_boost_round": 20,
+        "min_data": 10,
+        "num_leaves": 10,
+        "verbose": -1,
+    }
+
+    model1 = lgb.train(
+        params,
+        seq_ds,
+        keep_training_booster=True,
```

Comment:
```suggestion
```

Using `keep_training_booster=True` here works if the initial model (to be passed to `init_model`) is going to be passed onto training later **in memory in the same process**, but is that the situation that led to #6413.

I expect it will be more common to instead want to continue training with a model loaded from a file + a Sequence object in memory.

Could you please modify this test to not use `keep_training_booster=True`, or explain why it's necessary?

Reply:
Because I have a rolling timeseries trainning project. 
The dataset is too large to fit to the model. Aim to use memory efficiently, I iterately reading data, and build a model , make prediction for one month, then update datasource and retrain model.
Retrain model only use recent 12 months data, means old model with old weight would only be updated by recent data. 

```
for idx, (train_idx, test_idx) in enumerate(scroll_train_test(dates_partition, train_size=TRAIN_LOAD_STEP, test_size=TEST_LOAD_STEP, align_idx=train_end_idx)):
    
    train_partitions = dates_partition[train_idx]
    test_partitions = dates_partition[test_idx]

    train_df = read_partitioned_df(train_partitions, pre_train_partitions, train_df)
    test_df = read_partitioned_df(test_partitions, pre_train_partitions, test_df)
    ....
```

And another problem is

Reply:
Thank you for explaining that. Very interesting use of `Sequence`!

But the fact that **you** want to use this functionality in one specific way (with the model held in memory the entire time) does not mean that that's the only pattern that should be tested.

It's very common to use LightGBM's training continuation functionality starting from a model file... for example, to update an existing model once a month based on newly-arrived data. It's important that all LightGBM training-continuation codepaths support that pattern.

Anyway, like I mentioned in https://github.com/microsoft/LightGBM/pull/6414#discussion_r1683713834, I can push testing changes here. Once you see the diff of the changes I push, I'd be happy to answer any questions you have.